htdocs/core/class/commonobject.class.php

```diff
@@ -2543,16 +2543,14 @@ public function getChildrenOfLine($id)
 		$resql = $this->db->query($sql);
 		if ($resql)
 		{
-			$i = 0;
-			$num = $this->db->num_rows($resql);
-			while ($i < $num)
-			{
-				$row = $this->db->fetch_row($resql);
-				$rows[$i] = $row[0];
-				$i++;
-			}
-		}
+			if($this->db->num_rows($resql) > 0) {
+                while($row = $this->db->fetch_row($resql)) {
 
+                    $rows[] = $row[0];
+                    $rows = array_merge($rows, $this->getChildrenOfLine($row[0]));
```

Comment:
This seems to transform a function that return child into a function that return child + child of child, etc...If tru, to avoid breaking compatibility with current usage (first level child), you can add another parameter to the function, for exemple $includealltree = 0
PR must be pushed also on develop.